File: lib/common/alarm/alarm.dart
import 'dart:convert';

import 'package:json_annotation/json_annotation.dart';

part 'alarm.g.dart';

@JsonSerializable()
class Alarm {
  final int id;
  final int timestamp;
  final String? audioPath;
  final bool? fromAppAsset;
  final bool vibrate;
  final bool notification;
  final bool isAlarm;
  final bool loop;
  final int loopTimes;
  final String? notificationTitle;
  final String? notificationContent;

  Alarm({
    required this.id,
    required this.timestamp,
    this.audioPath,
    this.fromAppAsset,
    required this.vibrate,
    required this.notification,
    required this.isAlarm,
    required this.loop,
    required this.loopTimes,
    this.notificationTitle,
    this.notificationContent,
  });

  factory Alarm.fromJson(Map<String, dynamic> json) => _$AlarmFromJson(json);
  factory Alarm.fromJsonString(String json) => _$AlarmFromJson(jsonDecode(json));
  Map<String, dynamic> toJson() => _$AlarmToJson(this);
  String toJsonText() => jsonEncode(toJson());
}

File: android_bak/app/src/main/kotlin/cn/cutemc/pomotimer/pomotimer/alarm/Alarm.kt
package cn.cutemc.pomotimer.pomotimer.alarm

import kotlinx.serialization.Serializable
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json

@Serializable
data class Alarm(
    val id: Int,
    val timestamp: Long,
    val audioPath: String?,
    val fromAppAsset: Boolean?,
    val vibrate: Boolean,
    val loop: Boolean,
    val loopTimes: Int,
    val notificationTitle: String?,
    val notificationContent: String?,
) {
    companion object {
        fun fromJson(json: String): Alarm {
            return Json.decodeFromString<Alarm>(json)
        }
    }

    fun toJson(): String {
        return Json.encodeToString(this)
    }
}
